packages/node_modules/@webex/webex-core/src/lib/services/service-catalog.js

```diff
@@ -333,9 +333,15 @@ const ServiceCatalog = AmpState.extend({
         resolve();
       }
 
-      this.once(serviceGroup, () => resolve());
+      const timeoutTimer = setTimeout(() => reject(
+        new Error(`services: timeout occured while waiting for '${serviceGroup}' catalog to populate`)
+      ), (timeout) ? timeout * 1000 : 60000);
 
-      setTimeout(() => reject(), (timeout) ? timeout * 1000 : 60000);
+      clearTimeout(timeoutTimer);
+
+      this.once(serviceGroup, () => {
+        resolve();
```

Comment:
the `clearTimeout` function should go here. doing so outside the `once` would immediately clear it.